### ISSUE TITLE
Gradient accumulation: scale by actual microbatch count; log unscaled loss

### ISSUE THREAD

**Issue by jashvira** [jashvira/policy-gradient-experiments#5]
Problem
- Each microbatch loss is divided by config.gradient_accumulation_steps, but real microbatch count is ceil(batch_size/micro_batch_size).
- Logged loss aggregates scaled loss, not the true per-batch loss.

Impact
- Wrong gradient magnitude when config diverges; misleading metrics.

Code
```python
# grpo_microbatch_train_step(...):
scaled_loss = batch_loss / gradient_accumulation_steps
# caller passes config value, not actual count
```

Proposed fix
- Compute num_microbatches = ceil(batch_size/micro_batch_size) and pass into train step.
- Track/log unscaled batch_loss mean across microbatches separately from backprop-scaled loss.

Acceptance criteria
- Correct gradient scale regardless of slicing.
- Loss in logs equals mean per-example masked loss, independent of GA setting.

**Comment by jashvira** [comment]
Fixed in commit 43438f3: correct GA scaling by actual microbatch group size, per-group optimizer steps, and true unscaled loss logging. Meets Issue #5 acceptance criteria.
